libvast/vast/index_config.hpp

```diff
@@ -0,0 +1,53 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/type.hpp"
+
+#include <string>
+#include <vector>
+
+namespace vast {
+
+/// The configuration for `vast.index`.
+struct index_config {
+  static constexpr bool use_deep_to_string_formatter = true;
+
+  struct rule {
+    std::vector<std::string> targets;
```

Comment:
```suggestion
    std::vector<std::string> targets = {};
```

It's generally a good habit to initialize all member variables. Even though this change doesn't make any difference, it's easier to just always set the default rather than requiring developers to remember which types are trivially constructible and which are not.

Reply:
It feels excessive, but if this is the convention in the code base now, I'm going with it.